pkg/operator/starter.go

```diff
@@ -0,0 +1,133 @@
+)
+
+const (
+	operandName = "manila-csi-driver"
+
+	nfsImageEnvName = "NFS_DRIVER_IMAGE"
+)
+
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, "kube-client"))
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, util.OperatorNamespace, "")
+
+	// Create GenericOperatorclient. This is used by controllers created down below
+	gvr := operatorapi.SchemeGroupVersion.WithResource("clustercsidrivers")
+	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(controllerConfig.KubeConfig, gvr, string(operatorapi.ManilaCSIDriver))
+	if err != nil {
+		return err
+	}
+
+	manilaControllerSet := csicontrollerset.NewCSIControllerSet(
+		operatorClient,
+		controllerConfig.EventRecorder,
+	).WithLogLevelController().WithManagementStateController(
+		operandName,
+		false,
+	).WithStaticResourcesController(
+		"ManilaDriverStaticResources",
+		kubeClient,
+		kubeInformersForNamespaces,
+		generated.Asset,
+		[]string{
+			"csidriver.yaml",
+			"controller_sa.yaml",
+			"node_sa.yaml",
+			"rbac/snapshotter_binding.yaml",
+			"rbac/snapshotter_role.yaml",
+			"rbac/provisioner_binding.yaml",
+			"rbac/provisioner_role.yaml",
+			"rbac/privileged_role.yaml",
+			"rbac/controller_privileged_binding.yaml",
+			"rbac/node_privileged_binding.yaml",
+		},
+	).WithCSIDriverController(
+		"ManilaDriverController",
+		string(operatorapi.ManilaCSIDriver),
+		operandName,
+		util.OperatorNamespace,
+		assetWithNFSDriver,
+		kubeClient,
+		kubeInformersForNamespaces.InformersFor(util.OperatorNamespace),
+		csicontrollerset.WithControllerService("controller.yaml"),
+		csicontrollerset.WithNodeService("node.yaml"),
+	)
+
+	nfsController := csidrivercontroller.NewCSIDriverController(
+		"NFSDriverController",
+		string(operatorapi.ManilaCSIDriver),
+		"nfs-csi-driver",
+		util.OperatorNamespace,
+		operatorClient,
+		assetWithNFSDriver,
+		kubeClient,
+		controllerConfig.EventRecorder,
+	).WithNodeService(kubeInformersForNamespaces.InformersFor(util.OperatorNamespace).Apps().V1().DaemonSets(), "node-nfs.yaml")
+
+	openstackClient, err := manila.NewOpenStackClient(util.CloudConfigFilename, kubeInformersForNamespaces)
+	if err != nil {
+		return err
+	}
+	manilaController := manila.NewController(
+		operatorClient,
+		kubeClient,
+		kubeInformersForNamespaces,
+		openstackClient,
+		[]manila.Runnable{
+			manilaControllerSet,
+			nfsController,
+		},
+		controllerConfig.EventRecorder,
+	)
+	secretSyncController := secret.NewController(
+		operatorClient,
+		kubeClient,
+		kubeInformersForNamespaces,
+		controllerConfig.EventRecorder)
+
+	klog.Info("Starting the informers")
+	go kubeInformersForNamespaces.Start(ctx.Done())
+	go dynamicInformers.Start(ctx.Done())
+
+	klog.Info("Starting controllers")
+	go manilaController.Run(ctx, 1)
+	go secretSyncController.Run(ctx, 1)
```

Comment:
This controller will run regardless of manila; can we move it to the `manila.Runnable` slice above and have it run only when manila is present?

Reply:
Fixed